node/src/service/p2p/bootstrap.rs

```diff
@@ -162,140 +135,86 @@ pub(crate) async fn bootstrap(
-            }
-        }
-        other => debug!("Encountered other rendezvous event: {other:?}"),
-    }
-}
-
-/// Handles events within the gossipsub protocol
-fn on_gossipsub_event(
-    event: gossipsub::Event,
-    local_peer_id: PeerId,
-    registrations: &mut HashMap<PeerId, PeerInfo>,
-) {
-    match event {
-        // Received a message with peer information from another bootstrap node.
-        gossipsub::Event::Message {
-            propagation_source: peer_id,
-            message_id: id,
-            message,
-        } => {
-            // Got a message from ourselves, return early.
-            if peer_id == local_peer_id {
-                return;
+                SwarmEvent::Behaviour(BootstrapBehaviourEvent::RequestResponse(event)) => on_request_response_event(&mut swarm, event, ),
+                other => debug!("Encountered event: {other:?}"),
             }
-            // Deserialize peer info
-            let peer_info: PeerInfo = match serde_json::from_slice(&message.data) {
-                Ok(info) => info,
-                Err(..) => {
-                    error!(message:? = message.data; "Received invalid peer info from peer {peer_id:?}");
-                    return;
-                }
-            };
-            info!(
-                "Got registration: {:?} with id: {} from peer: {:?}",
-                peer_info, id, peer_id
-            );
-            insert_or_update_registrations(registrations, peer_info);
         }
-        other => debug!("Encountered other gossipsub event: {other:?}"),
     }
 }
 
 /// Handles events within the request_response protocol
 fn on_request_response_event(
     swarm: &mut Swarm<BootstrapBehaviour>,
     event: request_response::Event<PeerIdRequest, PeerInfoResponse>,
-    registrations: &HashMap<PeerId, PeerInfo>,
 ) {
     match event {
         // Message received, looking up the mapping
-        request_response::Event::Message { peer, message } => {
+        request_response::Event::Message { message, .. } => {
             if let Message::Request {
                 request,
                 channel,
                 request_id,
             } = message
             {
-                info!("Got request with id {request_id} from {peer}");
-                let id: PeerId = request.into();
-                let response = match registrations.get(&id) {
-                    Some(peer_info) => {
-                        info!("Peer {id:?} found in registrations");
-                        PeerInfoResponse::Found(peer_info.clone())
-                    }
-                    None => {
-                        info!("Peer {id:?} not found in registrations");
-                        PeerInfoResponse::NotFound(id.into())
-                    }
+                log::trace!("Received a request-response request ({request_id}): {request:?}");
+                // When we receive a message from
+                let Some(kref) = swarm.behaviour_mut().kad.kbucket(request.0) else {
```

Comment:
What if the current peer doesn't contain the information, but some other one does?
How would we query the network or the info **should be** in kademlia?

If I understand correctly, now we're just exposing addresses of servers which directly connected to the Collator.

Reply:
Great catch. I did that on purpose to minimize the amount of moving parts *right now*.

On failure to find we need to perform a get_closest_peers, which *should* add the peers it connects to, to the table. Once that query finishes, we search again